### DISCUSSION TITLE
Can anchor links be added to details?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#5143] [Q&A] [answered]
Given the fact that the insiders version of mkdocs-material allows to link to specific tabs of tabbed, I wonder if it is doable to achieve a similar behaviour with the details extension itself.

I use it for an FAQ page having the question in the summary and the answer in the body. The issue is, that it can't be linked to that easily, without adding a header which is visible in the toc and may look out of place in the details themself.

So now I'm looking for a way to have an anchor link to point to a details entry (And maybe even expand it?). Is that doable? And if so, how?

**Comment by squidfunk** [comment, marked as answer]
No, please use the search to find out why – this has already been discussed in depth.

**Reply by Andre601** [reply]
Okay. Bit sad it isn't possible as it limits my use-case quite a bit...

**Reply by facelessuser** [reply]
There will be a new way to create details and add IDs in Pymdown Extensions 9.10. I'm actually prepping a release. Any limitations outside that I cannot speak to.

**Reply by Andre601** [reply]
Okay. As long as it will allow direct linking to it through the anchor link will I be fine with it.

**Reply by facelessuser** [reply]
I do not control how linking is performed in Material, only that we will allow IDs to be easily placed on Details.

**Reply by facelessuser** [reply]
A release has been made.

**Reply by SPyofgame** [reply]
You can actually make an anchor and make the details appeared on the `TOC` using `###### <p hidden> something </p>`

With extra CSS, the anchor can be placed to the left border of the details conveniently without taking an empty space. (you can also move them up but I dont think it is good)

![image](https://user-images.githubusercontent.com/62207095/227764971-036892f8-d1f8-4458-b0c0-a0e8838978c2.png)

**Reply by SPyofgame** [reply]
![image](https://user-images.githubusercontent.com/62207095/228014311-1b6cb789-38a6-434b-841d-94ea9cae0206.png)
Oh seems like not a good idea, it can be incompatibility with [TOC] for some reason when it is inside a long admonitions
